9.0.1 deprecation in android build with sdk 33

> Task :geolocator_android:compileDebugJavaWithJavac FAILED
/Users/hesham/.pub-cache/hosted/pub.dartlang.org/geolocator_android-4.0.2/android/src/main/java/com/baseflow/geolocator/location/FusedLocationClient.java:199: warning: [deprecation] LocationRequest() in LocationRequest has been deprecated
    LocationRequest locationRequest = new LocationRequest();
                                      ^
/Users/hesham/.pub-cache/hosted/pub.dartlang.org/geolocator_android-4.0.2/android/src/main/java/com/baseflow/geolocator/location/FusedLocationClient.java:222: warning: [deprecation] PRIORITY_NO_POWER in LocationRequest has been deprecated
        return LocationRequest.PRIORITY_NO_POWER;
                              ^
/Users/hesham/.pub-cache/hosted/pub.dartlang.org/geolocator_android-4.0.2/android/src/main/java/com/baseflow/geolocator/location/FusedLocationClient.java:224: warning: [deprecation] PRIORITY_LOW_POWER in LocationRequest has been deprecated
        return LocationRequest.PRIORITY_LOW_POWER;
                              ^
/Users/hesham/.pub-cache/hosted/pub.dartlang.org/geolocator_android-4.0.2/android/src/main/java/com/baseflow/geolocator/location/FusedLocationClient.java:226: warning: [deprecation] PRIORITY_BALANCED_POWER_ACCURACY in LocationRequest has been deprecated
        return LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY;
                              ^
/Users/hesham/.pub-cache/hosted/pub.dartlang.org/geolocator_android-4.0.2/android/src/main/java/com/baseflow/geolocator/location/FusedLocationClient.java:228: warning: [deprecation] PRIORITY_HIGH_ACCURACY in LocationRequest has been deprecated
        return LocationRequest.PRIORITY_HIGH_ACCURACY;
                              ^
error: warnings found and -Werror specified
1 error
5 warnings

warnings found and -Werror specified

--- a/geolocator/example/android/app/build.gradle
+++ b/geolocator/example/android/app/build.gradle
@@ -21,6 +21,12 @@ if (flutterVersionName == null) {
     flutterVersionName = '1.0'
 }
 
+def args = ["-Xlint:deprecation", "-Xlint:unchecked", "-Werror"]
+
+project.getTasks().withType(JavaCompile) {
+    options.compilerArgs.addAll(args)
+}
+
 apply plugin: 'com.android.application'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 

--- a/geolocator/example/pubspec.yaml
+++ b/geolocator/example/pubspec.yaml
@@ -9,10 +9,8 @@ environment:
   sdk: ">=2.15.0 <3.0.0"
 
 dependencies:
-  baseflow_plugin_template:
-    git:
-      url: https://github.com/Baseflow/baseflow_plugin_template.git
-      ref: v2.1.0
+  baseflow_plugin_template: ^2.1.1
+  
   flutter:
     sdk: flutter
 

--- a/geolocator_android/CHANGELOG.md
+++ b/geolocator_android/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 4.1.1
+
+- Moves the Android complile arguments to the example `build.gradle` so that is not forces the arguments on consumers of the package.
+
 ## 4.1.0
 
 - Adds the ability to report altitude as MSL obtained from NMEA messages when available.

--- a/geolocator_android/android/build.gradle
+++ b/geolocator_android/android/build.gradle
@@ -1,6 +1,5 @@
 group 'com.baseflow.geolocator'
 version '1.0'
-def args = ["-Xlint:deprecation", "-Xlint:unchecked", "-Werror"]
 
 buildscript {
     repositories {
@@ -18,10 +17,6 @@ rootProject.allprojects {
         google()
         mavenCentral()
     }
-}
-
-project.getTasks().withType(JavaCompile) {
-    options.compilerArgs.addAll(args)
 }
 
 apply plugin: 'com.android.library'

--- a/geolocator_android/example/android/app/build.gradle
+++ b/geolocator_android/example/android/app/build.gradle
@@ -21,6 +21,12 @@ if (flutterVersionName == null) {
     flutterVersionName = '1.0'
 }
 
+def args = ["-Xlint:deprecation", "-Xlint:unchecked", "-Werror"]
+
+project.getTasks().withType(JavaCompile) {
+    options.compilerArgs.addAll(args)
+}
+
 apply plugin: 'com.android.application'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 

--- a/geolocator_android/example/pubspec.yaml
+++ b/geolocator_android/example/pubspec.yaml
@@ -9,10 +9,7 @@ environment:
   sdk: ">=2.15.0 <3.0.0"
 
 dependencies:
-  baseflow_plugin_template:
-    git:
-      url: https://github.com/Baseflow/baseflow_plugin_template.git
-      ref: v2.1.0
+  baseflow_plugin_template: ^2.1.1
   flutter:
     sdk: flutter
 

--- a/geolocator_android/pubspec.yaml
+++ b/geolocator_android/pubspec.yaml
@@ -2,7 +2,7 @@ name: geolocator_android
 description: Geolocation plugin for Flutter. This plugin provides the Android implementation for the geolocator.
 repository: https://github.com/baseflow/flutter-geolocator/tree/main/geolocator_android
 issue_tracker: https://github.com/baseflow/flutter-geolocator/issues?q=is%3Aissue+is%3Aopen
-version: 4.1.0
+version: 4.1.1
 
 environment:
   sdk: ">=2.15.0 <3.0.0"

--- a/geolocator_apple/example/pubspec.yaml
+++ b/geolocator_apple/example/pubspec.yaml
@@ -9,10 +9,7 @@ environment:
   sdk: ">=2.15.0 <3.0.0"
 
 dependencies:
-  baseflow_plugin_template:
-    git:
-      url: https://github.com/Baseflow/baseflow_plugin_template.git
-      ref: v2.1.0
+  baseflow_plugin_template: ^2.1.1
   flutter:
     sdk: flutter
 

--- a/geolocator_linux/example/pubspec.yaml
+++ b/geolocator_linux/example/pubspec.yaml
@@ -9,10 +9,7 @@ environment:
   sdk: ">=2.15.0 <3.0.0"
 
 dependencies:
-  baseflow_plugin_template:
-    git:
-      url: https://github.com/Baseflow/baseflow_plugin_template.git
-      ref: v2.1.0
+  baseflow_plugin_template: ^2.1.1
   flutter:
     sdk: flutter
 

--- a/geolocator_web/example/pubspec.yaml
+++ b/geolocator_web/example/pubspec.yaml
@@ -9,10 +9,7 @@ environment:
   sdk: ">=2.15.0-0 <3.0.0"
 
 dependencies:
-  baseflow_plugin_template:
-    git:
-      url: https://github.com/Baseflow/baseflow_plugin_template.git
-      ref: v2.1.0
+  baseflow_plugin_template: ^2.1.1
   flutter:
     sdk: flutter
 

--- a/geolocator_windows/example/pubspec.yaml
+++ b/geolocator_windows/example/pubspec.yaml
@@ -9,10 +9,7 @@ environment:
   sdk: ">=2.15.0 <3.0.0"
 
 dependencies:
-  baseflow_plugin_template:
-    git:
-      url: https://github.com/Baseflow/baseflow_plugin_template.git
-      ref: v2.1.0
+  baseflow_plugin_template: ^2.1.1
   flutter:
     sdk: flutter
 

